service/raft/raft_group0.hh

```diff
@@ -283,6 +283,11 @@ public:
         return _raft_gr.group0();
     }
 
+    // Returns a wrapper for the group0_server() with timeouts support.
+    raft_server_with_timeouts group0_server_with_timeouts() {
```

Comment:
So now we have `raft_server` and `raft_server_with_timeouts`, but `raft_server_with_timeouts` can get `nullopt` as a timeout and also do not have any timeouts. Why not have only one `raft_server`  that gets timeout and if no timeout is needed the caller will pass `nullopt` as a timeout parameter?

Reply:
> Why not have only one raft_server that gets timeout and if no timeout is needed the caller will pass nullopt as a timeout parameter?

don't need timeout -> pass `std::nullopt` as raft_timeout parameter
need specific timeout -> pass `raft_timeout{<specific_timeout_value>}` (not used right now)
need default for group timeout -> pass `raft_timeout{}`